extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -1194,11 +1200,44 @@ export class AmpLightboxGallery extends AMP.BaseElement {
   }
 
   /**
-   * Update thumbnails displayed in lightbox gallery.
+   * Update timestamps for all videos in gallery thumbnails.
+   * @private
+   */
+  updateVideoThumbnails_() {
+    const thumbnails = this.manager_.getThumbnails(this.currentLightboxGroupId_)
+        .map((thumbnail, index) => Object.assign({index}, thumbnail))
+        .filter(thumbnail => VIDEO_TAGS[thumbnail.element.tagName]);
+
+    this.vsync_.mutate(() => {
+      thumbnails.forEach(thumbnail => {
+        if (thumbnail.timestampPromise) {
+          thumbnail.timestampPromise.then(ts => {
+            // Many video players (e.g. amp-youtube) that don't support this API
+            // will often return 1. This is problematic for error checking.
+            if (!isNaN(ts)) {
```

Comment:
Prefer a shortcut return here to reduce nesting depth.

```js
if (isNaN(ts)) {
  return;
}
```

Reply:
Ack. @cvializ do you have an opinion on what to do about the fact that sometimes video players will return 1 when duration isn't supported? We could add an extra check for 1, but 1 is also actually a valid value for duration... Could we refactor the unsupported video players to return `-1` or `NaN`, or something that clearly isn't a valid duration? E.g. `<amp-youtube>`.

Reply:
I think a refactor of `1` as an empty value would be good idea.